mono/mini/aot-compiler.c

```diff
@@ -4832,7 +4832,9 @@ add_wrappers (MonoAotCompile *acfg)
 					g_assert (!strcmp (name, "ExportSymbol"));
 
 					/* load_cattr_value (), string case */
-					g_assert (*named != (char)0xff);
+MONO_DISABLE_WARNING (4310) // cast truncates constant value
+					g_assert (*named != (char)0xFF);
```

Comment:
Maybe cast pointer to ‘(const unsigned char*)’?

Reply:
Or guint8.

Reply:
And then remove the char cast.

Reply:
I experimented with similar casts in a different area a while back and got into some incorrect behaviors on some platform (don't recall what platform) failing some tests, so fallback to just disable the warning. I could of course retry the cast, but then I believe I should do it at the other places as well, since we at least have tests covering those code paths and failing if we don't get identical results on all platforms.

Reply:
"jaykrell feel free to revisit every `MONO_DISABLE_WARNING` for a 'better' fix, that neither quashes warnings nor changes semantics on any platform"?